### DISCUSSION TITLE
how to off toolbar and export to png/image

### DISCUSSION THREAD

**Discussion by rh-afk** [opengeos/leafmap#526] [Q&A] [answered]
Dear All,

how to off this toolbar from the map
![image](https://github.com/opengeos/leafmap/assets/141318843/4a8c33dc-428b-4a41-8d57-f2cbbea38a55)

and how to export the map as image and user don't need to know folder of our server because current practice is use the toolbar above, however user only can save in server folder and don't have ability to export in their own directory folder/computer.

thanks

**Comment by giswqs** [comment, marked as answer]
To hide the toolbar
```python
m = leafmap.Map(toolbar_control=False)
```

To allow users to download a file from the server:

```python
def create_download_link(filename, title="Click here to download: "):
    """Downloads a file from voila. Adopted from https://github.com/voila-dashboards/voila/issues/578

    Args:
        filename (str): The file path to the file to download
        title (str, optional): str. Defaults to "Click here to download: ".

    Returns:
        str: HTML download URL.
    """
    import base64

    from IPython.display import HTML

    data = open(filename, "rb").read()
    b64 = base64.b64encode(data)
    payload = b64.decode()
    basename = os.path.basename(filename)
    html = '<a download="{filename}" href="data:text/csv;base64,{payload}" style="color:#0000FF;" target="_blank">{title}</a>'
    html = html.format(payload=payload, title=title + f" {basename}", filename=basename)
    return HTML(html)
```

**Reply by rh-afk** [reply]
thanks @giswqs for how to off toolbar .
I mean in the map display we have option to download it as a file image like in plotly,altair,etc.We just simply need select which one the directory in our computer and save it . for example in here
![image](https://github.com/opengeos/leafmap/assets/141318843/5f6459a0-404d-4d28-87c1-4c79f796fa45)
is that possible to do that  in the leafmap?

**Reply by giswqs** [reply]
No. Because it is an interactive map, not a static image. Again, try ipyelaflet/folium directly. Leafmap supports anything that ipyleaflet/folium supports. Otherwise, leafmap probably does not support it either.

**Reply by rh-afk** [reply]
ok thanks @giswqs , one more question. How I can only switch on toolbar for layer selection?
![image](https://github.com/opengeos/leafmap/assets/141318843/6c1b3806-0191-439d-9b58-6733220d37c2)

thanks

**Reply by giswqs** [reply]
https://leafmap.org/leafmap/#leafmap.leafmap.Map.add_layer_manager
